src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts

```diff
@@ -71,8 +68,12 @@ export class FormDateTimePickerComponent implements ControlValueAccessor {
         }
     }
 
+    /**
+     * validates a date object
+     * @param date to be validated
+     */
     isValidDate(date: any) {
-        return date && Object.prototype.toString.call(date) === '[object Date]' && !isNaN(date);
+        return date && date instanceof Date && !isNaN(date.getDate());
```

Comment:
Why specifically `.getDate`?
It returns the day of month.

Reply:
This is needed to get a number value from the date, otherwise TS complains (TS2345: Argument of type 'Date' is not assignable to parameter of type 'number'.). Another way of doing this would be date.getTime(), both methods function in the same way and return NaN in case the date is invalid and otherwise a number.

Reply:
done :)